meta-ostro/conf/distro/include/ostroproject-ci.inc

```diff
@@ -31,27 +31,86 @@ SDKMACHINE ?= "x86_64"
 # Generate source mirror tarballs:
 BB_GENERATE_MIRROR_TARBALLS = "1"
 
-# Include the devel settings
+# If (and only if) building in the CI, the BUILD_ID is guaranteed to
+# be consistent across different builds and therefore it can and
+# should be included in the os-release package.
+#
+# When doing the same outside of the CI, we'd end up with an
+# os-release file that has BUILD_ID = ${DATETIME} for the first time
+# that the os-release file was generated. It does not get re-generated
+# (explicit vardepsexclude in os-release.bb) because that would cause
+# constant rebuilds.
+OS_RELEASE_FIELDS_append = " BUILD_ID"
```

Comment:
On Thu, 2016-03-31 at 08:16 -0700, Alexander D. Kanevskiy wrote:
> if you move version setting to OS_VERSION, there is no need for
> os-release.bb to know about BUILD_ID anymore.

OS_VERSION != BUILD_ID.

BUILD_ID is the full string, OS_VERSION is the reduced numeric version
used only by swupd images. It is valuable to record both in the
os-release file.

In the latest revision, "ostro-image" gets built again by the CI as a
image without swupd. In that image, OS_VERSION is not used at all, only
BUILD_ID is.